### DISCUSSION TITLE
Problem with axes_ranges

### DISCUSSION THREAD

**Discussion by j-foote** [pyvista/pyvista#5485] [Q&A]
I don't seem to be able to get "axes_ranges" to work with "show_grid".

For example:

```python
import numpy as np
import pyvista as pv

# Make up some data - gaussian
n = 20
x = np.linspace(-200, 200, num=n)
y = np.linspace(-200, 200, num=n)
xx, yy = np.meshgrid(x, y)
A, b = 50, 100
zz = A * np.exp(-0.5 * ((xx / b) ** 2.0 + (yy / b) ** 2.0))

s = 3
u, v = np.meshgrid(np.arange(n)*s, np.arange(n)*s)

# Make each bar
cubes = pv.MultiBlock()
for point in np.c_[u.reshape(-1), v.reshape(-1), zz.reshape(-1)]:
    c = point
    c[-1] /=2
    bar = pv.Cube(center=c, z_length=point[-1])
    cubes.append(bar)



# Plot all bars in 3D
p = pv.Plotter(notebook=0)
p.add_mesh(cubes, color='r', smooth_shading=False, show_edges=True)
p.show_grid(color='green', xtitle="magnitude", axes_ranges=[0, 10, 0, 10, 0, 10])
p.camera.azimuth = 35
p.camera.elevation = -10
p.show()
```

Shows:

<img width="392" alt="image" src="https://github.com/pyvista/pyvista/assets/98668157/375ac113-5166-44d4-9334-51194bfd9434">

My understanding is "axes_ranges=[0, 10, 0, 10, 0, 10]" should re-set the axis to run from 0 to 10, but it doesn't seem to work.

I'm using pyvista 0.43.2.

Is anyone else able to replicate this, or does anyone know what I'm doing wrong?

**Comment by JfosterBGC** [comment]
I am also having issues specifying the gridline values. 
I am trying to use the axes_range and the n_zbounds to get the Z to have grid lines at nice intervals (ie. 0, 25, 50, 75 etc.) 
I am rounding the Axes range and then determining the n_zbounds by dividing my range to get the interval. 

However, when I put those back into the show_grid, I just get the default. If I give it an n_zlabels value less than the default (i.e. 3 rather than 8), then it does reduce the number of grid lines but I can't seem to add more gridlines, and it is always some off intervals which is not matching my axes_range


```python
    g_spacing = 25
    axis_range = np.array(point_cloud.bounds)
    axis_range[4] = max(axis_range[4], 0) # Ensure Z min is not negative
    axis_range = np.round(axis_range/g_spacing)*g_spacing
    axis_intervals = axis_range.copy().reshape(3, 2)
    axis_intervals =  (axis_intervals[:,1]- axis_intervals[:,0])/g_spacing

    plotter.add_axes()
    plotter.show_grid(axes_ranges = axis_range, n_zlabels = axis_intervals.astype(np.int64)[2], show_zaxis = True)
```
